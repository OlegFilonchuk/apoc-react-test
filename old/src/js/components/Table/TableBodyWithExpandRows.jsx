import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { isEqual } from 'lodash';

import './TableBodyWithExpandRows.less';

/* eslint-disable no-shadow */
export default class TableBodyWithExpandRows extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      expandedRowId: props.defaultExpandedRowId,
      expandedChildren: (props.isControlled && props.expandedChildren) || []
    };
  }

  componentWillReceiveProps(nextProps) {
    const child = React.Children.toArray(nextProps.children).find(child => child.props.isExpanded);

    if (child && child.props.id !== this.state.expandedRowId) {
      this.setState({ expandedRowId: child.props.id });
    }

    if (this.props.isControlled && !isEqual(nextProps.expandedChildren, this.props.expandedChildren)) {
      this.setState({ expandedChildren: nextProps.expandedChildren });
    }
  }

  /**
   * It should be child's prop id
   * @param {string} id
   */
  onExpand = id => {
    const { expandedRowId, expandedChildren } = this.state;
    const { allowMultipleExpanded } = this.props;

    if (!allowMultipleExpanded) {
      this.setState({ expandedRowId: expandedRowId === id ? null : id });
    }

    const child = React.Children.toArray(this.props.children).find(child => child.props.id === id);

    if (child.props.onExpand && !allowMultipleExpanded) {
      child.props.onExpand(child.props.id);
    } else if (allowMultipleExpanded && !expandedChildren.includes(child.props.id)) {
      this.setState({
        expandedChildren: [...expandedChildren, child.props.id]
      });
    } else if (allowMultipleExpanded && expandedChildren.includes(child.props.id)) {
      this.setState({
        expandedChildren: expandedChildren.filter(childId => childId !== child.props.id)
      });
    }
    if (this.props.isControlled) {
      this.props.onExpand(id);
    }
  };

  get tableBodyChildren() {
    return React.Children.map(this.props.children, child => {
      const cells = React.Children.toArray(child.props.children);
      const colSpan = this.props.customColspan ? this.props.customColspan : React.Children.count(cells);

      return [
        React.cloneElement(child, {
          className: this.getChildClassNames(child.props.id, child.props.className),
          onExpand: this.onExpand,
          isExpanded: this.isChildExpanded(child.props.id),
          expandOnClick: this.props.expandOnClick
        }),
        this.isChildExpanded(child.props.id) && (
          <tr className="expandable-table-row-content">
            <td colSpan={colSpan}>{child.props.expandContent}</td>
          </tr>
        )
      ];
    });
  }

  get classNames() {
    return classNames('table-body-with-expand-content', this.props.className);
  }

  getChildClassNames = (childId, childClassName) =>
    classNames('expandable-table-row-trigger', childClassName, {
      'expandable-table-row-trigger--expanded': this.isChildExpanded(childId)
    });

  isChildExpanded = childId =>
    !this.props.allowMultipleExpanded
      ? this.state.expandedRowId === childId
      : this.state.expandedChildren.includes(childId);

  render() {
    return this.props.customWrapper ? (
      this.props.customWrapper(props => (
        <tbody {...props} ref={props.innerRef} className={this.classNames}>
          {this.tableBodyChildren}
        </tbody>
      ))
    ) : (
      <tbody className={this.classNames}>{this.tableBodyChildren}</tbody>
    );
  }
}

TableBodyWithExpandRows.propTypes = {
  /**
   * List of custom classes added to component
   */
  className: PropTypes.string,

  /**
   * Array of <tr> elements
   */
  children: PropTypes.node,

  /**
   * TableRowWithExpandContent's id which should be expanded by default
   */
  defaultExpandedRowId: PropTypes.string,

  /**
   * If true a whole row will be clickable
   */
  expandOnClick: PropTypes.bool,

  /**
   * If true, rows don't be collapsed automatically
   */
  allowMultipleExpanded: PropTypes.bool,

  /**
   * Possibility to set custom number of colspan
   */
  customColspan: PropTypes.number,

  /**
   * Custom wrapper for table children. By default, table component and dependant expects particular children types
   * as they are operating on them. This props allow to add custom wrapper around them AFTER they are modified.
   */
  customWrapper: PropTypes.func,

  /**
   * Allow to controll collapse/expadned state from outside of component
   */
  isControlled: PropTypes.bool,
  /**
   * Function which allow to change outside collapse/expanded state. It gets id of clicked element.
   */
  onExpand: PropTypes.func,
  /**
   * String with id of element which is currently expanded (this should be passed only if isControlled is true and allowMultipleExpandedState is false)
   */
  expandedRowId: PropTypes.string,
  /**
   * Array of currently expanded elements ids in table (this should be passed only if isControlled and allowMultipleExpandedState are true)
   */
  expandedChildren: PropTypes.arrayOf(PropTypes.string)
};

TableBodyWithExpandRows.defaultProps = {
  defaultExpandedRowId: null,
  className: '',
  children: [],
  expandOnClick: false,
  allowMultipleExpanded: false,
  customColspan: null,
  customWrapper: null,
  isControlled: false,
  onExpand: () => {},
  expandedRowId: '',
  expandedChildren: []
};
