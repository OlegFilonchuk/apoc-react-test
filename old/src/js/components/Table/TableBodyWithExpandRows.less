@import '~chameleon/less/variables.less';
@import './tableVariables.less';

.table-body-with-expand-content {
  .expandable-table-row-content {
    padding: 5px;

    > td {
      background-color: @white-color;
      border-bottom: 1px solid @tableBorderColor;
    }
  }
}
