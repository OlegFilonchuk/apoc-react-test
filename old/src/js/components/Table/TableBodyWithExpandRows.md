## Necessary sub component is
* [TableRowWithExpandContent](#tablerowwithexpandcontent)

## Example of table with expand row
It uses [TableHeadRowForExpandRows](#tableheadrowforexpandrows), [TableBodyWithExpandRows](#tablebodywithexpandrows) and [TableRowWithExpandContent](#tablerowwithexpandcontent)

    (
      <Table>
        <TableHeadRowForExpandRows>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRowForExpandRows>
        <TableBodyWithExpandRows>
          <TableRowWithExpandContent id="id-1" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent id="id-2" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent id="id-3" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
        </TableBodyWithExpandRows>
      </Table>
    )

## Example of table with expanded row by default
It uses [TableHeadRowForExpandRows](#tableheadrowforexpandedrows), [TableBodyWithExpandRows](#tablebodywithexpandrows) and [TableRowWithExpandContent](#tablerowwithexpandcontent)

    (
      <Table>
        <TableHeadRowForExpandRows>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRowForExpandRows>
        <TableBodyWithExpandRows defaultExpandedRowId="id-2">
          <TableRowWithExpandContent id="id-1" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent id="id-2" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent id="id-3" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
        </TableBodyWithExpandRows>
      </Table>
    )
    
    
## Example of table with clickable rows
It uses [TableHeadRowForExpandRows](#tableheadrowforexpandedrows), [TableBodyWithExpandRows](#tablebodywithexpandrows) and [TableRowWithExpandContent](#tablerowwithexpandcontent)
        
    (
      <Table isRowsStriped>
        <TableHeadRowForExpandRows>
          <th width="20%">Title</th>
          <th width="20%">Title</th>
          <th>Title</th>
        </TableHeadRowForExpandRows>
        <TableBodyWithExpandRows expandOnClick>
          <TableRowWithExpandContent
              id="id-1"
              expandContent="hello world"
          >
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent
              id="id-2"
              expandContent="hello world"
          >
            <td>Content</td>
            <td>Content</td>
            <TableCellWithTextEllipsis>No ellipsis should be turned on</TableCellWithTextEllipsis>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent
              id="id-3"
              expandContent="hello world"
          >
            <TableCellWithTextEllipsis>This content should have some ellipis on!</TableCellWithTextEllipsis>
            <td>Content</td>
            <TableCellWithTextEllipsis>
            This content should have some ellipis on!This content should have some ellipis on!This content should have some ellipis on!
            This content should have some ellipis on!This content should have some ellipis on!This content should have some ellipis on!
            This content should have some ellipis on!This content should have some ellipis on!This content should have some ellipis on!
            This content should have some ellipis on!This content should have some ellipis on!This content should have some ellipis on!
            </TableCellWithTextEllipsis>
          </TableRowWithExpandContent>
        </TableBodyWithExpandRows>
      </Table>
    )

## Example of table with clickable rows - buttons blocks row expansion 
It uses [TableHeadRowForExpandRows](#tableheadrowforexpandedrows), [TableBodyWithExpandRows](#tablebodywithexpandrows) and [TableRowWithExpandContent](#tablerowwithexpandcontent)

    onClick = ()=> null;        
    (
      <Table isRowsStriped>
        <TableHeadRowForExpandRows>
          <th width="20%">Title</th>
          <th width="20%">Title</th>
          <th>Title</th>
          <th>Actions</th>
        </TableHeadRowForExpandRows>
        <TableBodyWithExpandRows expandOnClick>
          <TableRowWithExpandContent
              id="id-1"
              expandContent="hello world"
          >
            <td>Content</td>
            <td><a href="https://google.com" target="_blank" referrerPolicy="no-referrer">Clicking on me won't expands row</a></td>
            <td><button>Clicking on me wont expand row</button></td>
            <td>
              <ContextMenu>
                <ContextOptionLink onClick={() => { onClick('ContextOptionLink clicked!'); }}>Edit</ContextOptionLink>
                <ContextOption onClick={() => { onClick('ContextOption "Rename" clicked!'); }}>Rename</ContextOption>
              </ContextMenu>
            </td>
          </TableRowWithExpandContent>        
        </TableBodyWithExpandRows>
      </Table>
    )

## Example of table with multiple expand rows 
It uses [TableHeadRowForExpandRows](#tableheadrowforexpandrows), [TableBodyWithExpandRows](#tablebodywithexpandrows) and [TableRowWithExpandContent](#tablerowwithexpandcontent)

    (
      <Table>
        <TableHeadRowForExpandRows>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRowForExpandRows>
        <TableBodyWithExpandRows allowMultipleExpanded>
          <TableRowWithExpandContent id="id-1" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent id="id-2" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
          <TableRowWithExpandContent id="id-3" expandContent="hello world">
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRowWithExpandContent>
        </TableBodyWithExpandRows>
      </Table>
    )