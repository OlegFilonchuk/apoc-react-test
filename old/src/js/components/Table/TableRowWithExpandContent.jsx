import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import TableCellWithTextEllipsis from './TableCellWithTextEllipsis';
import ExpandButton from './ExpandButton';

import './TableRowWithExpandContent.less';

const emptyFn = () => null;

const clickableTags = ['BUTTON', 'A'];

function isClickableElementClicked(target, stopAtTagName) {
  let currentTarget = target;

  while (currentTarget.tagName !== stopAtTagName) {
    if (clickableTags.includes(currentTarget.tagName) && !currentTarget.classList.contains('expand-btn')) {
      return true;
    }

    currentTarget = currentTarget.parentNode;
  }

  return false;
}

export default class TableRowWithExpandContent extends React.Component {
  onExpand = e => {
    const target = e.nativeEvent.target;

    if (isClickableElementClicked(target, 'TR')) {
      return;
    }

    this.props.onExpand(this.props.id);
  };

  get expandBtn() {
    return (
      <ExpandButton onExpand={this.props.expandOnClick ? emptyFn : this.onExpand} isExpanded={this.props.isExpanded} />
    );
  }

  getCellForExpandedRow = cell => {
    const { children, className, ...rest } = cell.props;

    return this.props.isExpanded && cell.type === TableCellWithTextEllipsis ? (
      <td className={classNames('table-cell-with-expand-btn', className)} {...rest}>
        {this.expandBtn}
        {children}
      </td>
    ) : (
      React.cloneElement(cell, {
        className: classNames('table-cell-with-expand-btn', cell.props.className),
        children: React.Children.map(cell.props.children, element => [this.expandBtn, element])
      })
    );
  };

  get finalClassName() {
    return classNames(this.props.className, {
      'table-row-with-expand-content__clickable': this.props.expandOnClick
    });
  }

  getFinalContent = ({ columnWithButton, content, additionalProps = {} }) => (
    <tr
      {...additionalProps}
      className={this.finalClassName}
      onClickCapture={this.props.expandOnClick ? this.onExpand : emptyFn}
    >
      {columnWithButton}
      {content}
    </tr>
  );

  render() {
    const [firstColumn, ...restColumns] = React.Children.toArray(this.props.children);
    const columnWithButton = this.getCellForExpandedRow(firstColumn);
    const columns = [...restColumns];
    const content = React.Children.map(React.Children.toArray(columns), child =>
      this.props.isExpanded && child.type === TableCellWithTextEllipsis ? <td {...child.props} /> : child
    );

    return this.props.customWrapper
      ? this.props.customWrapper(props =>
          this.getFinalContent({
            columnWithButton,
            content,
            additionalProps: {
              ...props,
              ref: props.innerRef
            }
          })
        )
      : this.getFinalContent({ columnWithButton, content });
  }
}

TableRowWithExpandContent.propTypes = {
  /* eslint-disable react/no-unused-prop-types */
  /**
   * TableRowWithExpandContent's id
   */
  id: PropTypes.string.isRequired,

  /**
   * It will be called if expand button is clicked
   */
  onExpand: PropTypes.func,
  /* eslint-enable react/no-unused-prop-types */

  /**
   * List of custom classes added to component
   */
  className: PropTypes.string,

  /**
   * Array of table's cells
   */
  children: PropTypes.node.isRequired,

  /**
   * Show if row was expanded - used only by TableBodyWithExpandRows component internally
   */
  isExpanded: PropTypes.bool,

  /**
   * if true it triggers onExpand prop - used only by TableBodyWithExpandRows component internally
   */
  expandOnClick: PropTypes.bool,

  /**
   * Custom wrapper for table children. By default, table component and dependant expects particular children types
   * as they are operating on them. This props allow to add custom wrapper around them AFTER they are modified.
   */
  customWrapper: PropTypes.func
};

TableRowWithExpandContent.defaultProps = {
  className: '',
  onClick: null,
  isExpanded: false,
  expandOnClick: false,
  onExpand: () => null,
  customWrapper: null
};
