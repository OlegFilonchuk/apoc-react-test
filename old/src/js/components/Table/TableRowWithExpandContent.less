@import './tableVariables.less';
@import '~chameleon/less/variables';

@paddingLeftForCellWithExpandButton: calc(@expandButtonWidth + @cellPadding);

.expandable-table-row-trigger {
  &.expandable-table-row-trigger--expanded {
    td {
      background-color: @emptyTableContainerBackground;
    }
  }

  .table-cell-with-expand-btn {
    position: relative;
    padding-left: @paddingLeftForCellWithExpandButton;

    .expand-content-wrap {
      padding-left: @paddingLeftForCellWithExpandButton;
    }
  }

  .expand-btn {
    background: none;
    border: none;
    border-radius: 0;
    border-right: 1px solid @tableBorderColor;
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: @expandButtonWidth;
    z-index: 2;
  }
}

.table-row-with-expand-content__clickable {
  cursor: pointer;

  &:hover > td {
    background-color: @table-bg-hover;
  }
}
