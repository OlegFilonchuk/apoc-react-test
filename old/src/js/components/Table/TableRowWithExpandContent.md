Purpose of this component is to be used together with [TableBodyWithExpandRows](#tablehbodywithexpandrows). See example there.
