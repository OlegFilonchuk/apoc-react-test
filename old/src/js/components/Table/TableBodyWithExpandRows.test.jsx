import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import TableBodyWithExpandRows from './TableBodyWithExpandRows';
import TableRowWithExpandContent from './TableRowWithExpandContent';

const mountBody = ({ mountMethod = mount, props = {} }) =>
  mountMethod(
    <TableBodyWithExpandRows {...props}>
      <TableRowWithExpandContent isExpanded={false} id="id-1" expandContent="Hello world">
        <td>Content</td>
      </TableRowWithExpandContent>
      <TableRowWithExpandContent isExpanded={false} id="id-2" expandContent="Hello world">
        <td>Content</td>
      </TableRowWithExpandContent>
    </TableBodyWithExpandRows>,
    {
      attachTo: document.createElement('table')
    }
  );

describe('<TableBodyWithExpandRows />', () => {
  describe('should match snapshot when row ', () => {
    it('is not expanded', () => {
      expect(toJson(mountBody({ mountMethod: shallow }))).toMatchSnapshot();
    });

    it('is expanded', () => {
      expect(toJson(mountBody({ mountMethod: shallow, props: { defaultExpandedRowId: 'id-1' } }))).toMatchSnapshot();
    });
  });

  describe('correctly set `className`', () => {
    let body;

    beforeEach(() => {
      body = mountBody({});
    });

    it('without passing `className`', () => {
      const element = body.find('.table-body-with-expand-content');

      expect(element.length).toBe(1);
      expect(element).toHaveProp('className', 'table-body-with-expand-content');
    });

    it('with passing `className`', () => {
      expect(body.setProps({ className: 'some-extra-class' })).toHaveClassName('some-extra-class');
    });
  });

  describe('if `defaultExpandedRowId` props', () => {
    it('was passed content should be expanded', () => {
      const body = mountBody({ props: { defaultExpandedRowId: 'id-1' } });
      const elements = body.find('.expandable-table-row-trigger');

      expect(elements.at(0)).toHaveClassName('expandable-table-row-trigger expandable-table-row-trigger--expanded');
      expect(body.find('tr.expandable-table-row-trigger--expanded').length).toBe(1);
      expect(body.find('tr.expandable-table-row-content').length).toBe(1);
    });

    it("wasn't passed content shouldn't be expanded", () => {
      const body = mountBody({});

      expect(body.find('.expandable-table-row-trigger--expanded').length).toBe(0);
      expect(body.find('.expandable-table-row-content').length).toBe(0);
    });
  });

  it('expand row on expand button click', () => {
    const body = mountBody({});
    const button = body.find('.expand-btn').at(0);

    button.simulate('click');

    expect(body.find('tr.expandable-table-row-trigger--expanded').length).toBe(1);
    expect(body.find('.expandable-table-row-content').length).toBe(1);
  });

  describe('when children is not a standard td', () => {
    let CustomTD;
    let errorConsole;
    let mountedElement;

    beforeAll(() => {
      errorConsole = spyOn(console, 'error');
      CustomTD = () => <td>I am a Custom TD</td>;

      mountedElement = mount(
        <TableBodyWithExpandRows defaultExpandedRowId="test">
          <TableRowWithExpandContent isExpanded id="test" expandContent="Hello world">
            <CustomTD />
            <CustomTD />
            <CustomTD />
          </TableRowWithExpandContent>
        </TableBodyWithExpandRows>,
        {
          attachTo: document.createElement('table')
        }
      );
    });

    it('matches snapshot', () => {
      expect(toJson(mountedElement)).toMatchSnapshot();
    });

    it('TD for expanded content has proper colSpan value', () => {
      const tdFromExpandedContent = mountedElement.find('.expandable-table-row-content > td');

      expect(tdFromExpandedContent).toBePresent();
      expect(tdFromExpandedContent.prop('colSpan')).toEqual(3);
    });

    it('prints no errors', () => {
      expect(errorConsole).not.toHaveBeenCalled();
    });
  });

  describe('when children is not a standard td and false values', () => {
    let CustomTD;
    let errorConsole;
    let mountedElement;

    beforeAll(() => {
      errorConsole = spyOn(console, 'error');
      CustomTD = () => <td>I am a Custom TD</td>;

      mountedElement = mount(
        <TableBodyWithExpandRows defaultExpandedRowId="test">
          <TableRowWithExpandContent isExpanded id="test" expandContent="Hello world">
            <CustomTD />
            <CustomTD />
            {false}
            <CustomTD />
            {false && <CustomTD />}
          </TableRowWithExpandContent>
        </TableBodyWithExpandRows>,
        {
          attachTo: document.createElement('table')
        }
      );
    });

    it('matches snapshot', () => {
      expect(toJson(mountedElement)).toMatchSnapshot();
    });

    it('TD for expanded content has proper colSpan value 3, ignoring Boolean values elements', () => {
      const tdFromExpandedContent = mountedElement.find('.expandable-table-row-content > td');

      expect(tdFromExpandedContent).toBePresent();
      expect(tdFromExpandedContent.prop('colSpan')).toEqual(3);
    });

    it('prints no errors', () => {
      expect(errorConsole).not.toHaveBeenCalled();
    });
  });
});
