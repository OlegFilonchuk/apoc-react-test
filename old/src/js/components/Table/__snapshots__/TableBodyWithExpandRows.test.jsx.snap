// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableBodyWithExpandRows /> should match snapshot when row  is expanded 1`] = `
<tbody
  className="table-body-with-expand-content"
>
  <TableRowWithExpandContent
    className="expandable-table-row-trigger expandable-table-row-trigger--expanded"
    customWrapper={null}
    expandContent="Hello world"
    expandOnClick={false}
    id="id-1"
    isExpanded={true}
    key=".0/.0"
    onClick={null}
    onExpand={[Function]}
  >
    <td>
      Content
    </td>
  </TableRowWithExpandContent>
  <tr
    className="expandable-table-row-content"
    key=".0/.1"
  >
    <td
      colSpan={1}
    >
      Hello world
    </td>
  </tr>
  <TableRowWithExpandContent
    className="expandable-table-row-trigger"
    customWrapper={null}
    expandContent="Hello world"
    expandOnClick={false}
    id="id-2"
    isExpanded={false}
    key=".1/.0"
    onClick={null}
    onExpand={[Function]}
  >
    <td>
      Content
    </td>
  </TableRowWithExpandContent>
</tbody>
`;

exports[`<TableBodyWithExpandRows /> should match snapshot when row  is not expanded 1`] = `
<tbody
  className="table-body-with-expand-content"
>
  <TableRowWithExpandContent
    className="expandable-table-row-trigger"
    customWrapper={null}
    expandContent="Hello world"
    expandOnClick={false}
    id="id-1"
    isExpanded={false}
    key=".0/.0"
    onClick={null}
    onExpand={[Function]}
  >
    <td>
      Content
    </td>
  </TableRowWithExpandContent>
  <TableRowWithExpandContent
    className="expandable-table-row-trigger"
    customWrapper={null}
    expandContent="Hello world"
    expandOnClick={false}
    id="id-2"
    isExpanded={false}
    key=".1/.0"
    onClick={null}
    onExpand={[Function]}
  >
    <td>
      Content
    </td>
  </TableRowWithExpandContent>
</tbody>
`;

exports[`<TableBodyWithExpandRows /> when children is not a standard td and false values matches snapshot 1`] = `
<TableBodyWithExpandRows
  allowMultipleExpanded={false}
  className=""
  customColspan={null}
  customWrapper={null}
  defaultExpandedRowId="test"
  expandOnClick={false}
  expandedChildren={Array []}
  expandedRowId=""
  isControlled={false}
  onExpand={[Function]}
>
  <tbody
    className="table-body-with-expand-content"
  >
    <TableRowWithExpandContent
      className="expandable-table-row-trigger expandable-table-row-trigger--expanded"
      customWrapper={null}
      expandContent="Hello world"
      expandOnClick={false}
      id="test"
      isExpanded={true}
      key=".0/.0"
      onClick={null}
      onExpand={[Function]}
    >
      <tr
        className="expandable-table-row-trigger expandable-table-row-trigger--expanded"
        onClickCapture={[Function]}
      >
        <CustomTD
          className="table-cell-with-expand-btn"
          key=".0"
        >
          <td>
            I am a Custom TD
          </td>
        </CustomTD>
        <CustomTD
          key=".$.$.1"
        >
          <td>
            I am a Custom TD
          </td>
        </CustomTD>
        <CustomTD
          key=".$.$.3"
        >
          <td>
            I am a Custom TD
          </td>
        </CustomTD>
      </tr>
    </TableRowWithExpandContent>
    <tr
      className="expandable-table-row-content"
      key=".0/.1"
    >
      <td
        colSpan={3}
      >
        Hello world
      </td>
    </tr>
  </tbody>
</TableBodyWithExpandRows>
`;

exports[`<TableBodyWithExpandRows /> when children is not a standard td matches snapshot 1`] = `
<TableBodyWithExpandRows
  allowMultipleExpanded={false}
  className=""
  customColspan={null}
  customWrapper={null}
  defaultExpandedRowId="test"
  expandOnClick={false}
  expandedChildren={Array []}
  expandedRowId=""
  isControlled={false}
  onExpand={[Function]}
>
  <tbody
    className="table-body-with-expand-content"
  >
    <TableRowWithExpandContent
      className="expandable-table-row-trigger expandable-table-row-trigger--expanded"
      customWrapper={null}
      expandContent="Hello world"
      expandOnClick={false}
      id="test"
      isExpanded={true}
      key=".0/.0"
      onClick={null}
      onExpand={[Function]}
    >
      <tr
        className="expandable-table-row-trigger expandable-table-row-trigger--expanded"
        onClickCapture={[Function]}
      >
        <CustomTD
          className="table-cell-with-expand-btn"
          key=".0"
        >
          <td>
            I am a Custom TD
          </td>
        </CustomTD>
        <CustomTD
          key=".$.$.1"
        >
          <td>
            I am a Custom TD
          </td>
        </CustomTD>
        <CustomTD
          key=".$.$.2"
        >
          <td>
            I am a Custom TD
          </td>
        </CustomTD>
      </tr>
    </TableRowWithExpandContent>
    <tr
      className="expandable-table-row-content"
      key=".0/.1"
    >
      <td
        colSpan={3}
      >
        Hello world
      </td>
    </tr>
  </tbody>
</TableBodyWithExpandRows>
`;
